src/Controller/HandleApprovalController.php

```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types=1);
+
+namespace Pdsinterop\Solid\Controller;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class HandleApprovalController extends ServerController
+{    
+    public function __invoke(ServerRequestInterface $request, array $args) : ResponseInterface
+    {
+		$clientId = $args['clientId'];
+		$returnUrl = $_POST['returnUrl'];
```

Comment:
For future reference: POST parameters can be retrieved from a PSR7 Request through `$request->getParsedBody()`